### PR TITLE
Fixed bug in setup.py and made coreg + spatial_tools tests pass

### PR DESCRIPTION
**Pull request by erikmannerfelt**
The `setup.py` file still had a reference to DemUtils, so `pip install -e .` failed for me before fixing it.

Some small fixes in `tests/test_coreg.py` and `tests/test_spatial_tools.py` made them pass on my machine.

`tests/test_dem.py` still fails for me with the error:

```bash
    def __parse_metadata_from_fn(self,silent=False):
    
        """
        Attempts to pull metadata (e.g., sensor, date information) from fname, setting sensor, satellite,
        tile, datetime, and date attributes.
        """
    
        fname = self.filename
        name_attrs = ['satellite', 'sensor', 'product', 'version', 'tile_name', 'datetime']
        attrs = parse_metadata_from_fn(fname)
    
        for n in name_attrs:
            a = self.__getattribute__(n)
>           a_fn =  attrs[name_attrs.index(n)]
E           TypeError: 'generator' object is not subscriptable

/home/erik/.local/share/conda/xdem/lib/python3.8/site-packages/geoutils/satimg.py:305: TypeError
---------------------------------------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------------------------------------
No metadata could be read from filename.
========================================================================================================= short test summary info =========================================================================================================
FAILED tests/test_dem.py::TestDEM::test_load_DEM_subclass - TypeError: 'generator' object is not subscriptable

```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,7 +13,7 @@
       license='BSD-3',
       packages=['xdem'],
       install_requires=['numpy', 'scipy', 'rasterio', 'geopandas', 'pyproj'],
-      extras_require={'rioxarray': ['rioxarray'], 'richdem': ['richdem'], 'pdal':['pdal']},
+      extras_require={'rioxarray': ['rioxarray'], 'richdem': ['richdem'], 'pdal': ['pdal']},
```
Comment (adehecq) [inline setup.py:16]: tqdm is also required to run the tests

Comment (erikmannerfelt) [reply setup.py:16]: Not just the tests, actually, but the entire `coreg.py`

**Comment by adehecq** [comment]
I pulled your branch Erik and I also get several errors when running the tests:
- first is that the path to the test data should be set with absolute paths. To check if that works, I suggest installing the package without the -e option and run from outside the package folder. That's when issues start...
- in test_vref, I get 

>           geoid=pyproj.Proj(init="EPSG:4326", geoidgrids='us_nga_egm08_25.tif')
> 
> tests/test_dem.py:50: 
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
> ../../../miniconda3/envs/work/lib/python3.8/site-packages/pyproj/proj.py:171: in __init__
>     super(Proj, self).__init__(cstrencode(projstring.strip()))
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
> 
> >   ???
> E   pyproj.exceptions.ProjError: Invalid projection b'+proj=longlat +datum=WGS84 +geoidgrids=us_nga_egm08_25.tif +vunits=m +no_defs'.: (Internal Proj Error: proj_create: Error -38: failed to load datum shift file)
> 
> pyproj/_proj.pyx:30: ProjError
>

File: setup.py (hunk 1 above)
Comment (erikmannerfelt) [reply setup.py:16]: Added it as a dependency in 8c76cbe

**Comment by erikmannerfelt** [comment]
> I pulled your branch Erik and I also get several errors when running the tests:
> 
>     * first is that the path to the test data should be set with absolute paths. To check if that works, I suggest installing the package without the -e option and run from outside the package folder. That's when issues start...
> 
>     * in test_vref, I get
> 
> 
> > ```
> >       geoid=pyproj.Proj(init="EPSG:4326", geoidgrids='us_nga_egm08_25.tif')
> > ```
> > 
> > 
> > tests/test_dem.py:50:
> > 
> > ../../../miniconda3/envs/work/lib/python3.8/site-packages/pyproj/proj.py:171: in **init**
> > super(Proj, self).**init**(cstrencode(projstring.strip()))
> > 
> > > ???
> > > E   pyproj.exceptions.ProjError: Invalid projection b'+proj=longlat +datum=WGS84 +geoidgrids=us_nga_egm08_25.tif +vunits=m +no_defs'.: (Internal Proj Error: proj_create: Error -38: failed to load datum shift file)
> > 
> > 
> > pyproj/_proj.pyx:30: ProjError

@adehecq  what version of proj do you have? I've had the exact same issue using `proj=7.1.0` but not in `proj=7.2.0`

**Comment by adehecq** [comment]
> > I pulled your branch Erik and I also get several errors when running the tests:
> > ```
> > * first is that the path to the test data should be set with absolute paths. To check if that works, I suggest installing the package without the -e option and run from outside the package folder. That's when issues start...
> > 
> > * in test_vref, I get
> > ```
> > 
> > 
> > > ```
> > >       geoid=pyproj.Proj(init="EPSG:4326", geoidgrids='us_nga_egm08_25.tif')
> > > ```
> > > 
> > > 
> > > tests/test_dem.py:50:
> > > ../../../miniconda3/envs/work/lib/python3.8/site-packages/pyproj/proj.py:171: in **init**
> > > super(Proj, self).**init**(cstrencode(projstring.strip()))
> > > > ???
> > > > E   pyproj.exceptions.ProjError: Invalid projection b'+proj=longlat +datum=WGS84 +geoidgrids=us_nga_egm08_25.tif +vunits=m +no_defs'.: (Internal Proj Error: proj_create: Error -38: failed to load datum shift file)
> > > 
> > > 
> > > pyproj/_proj.pyx:30: ProjError
> 
> @adehecq what version of proj do you have? I've had the exact same issue using `proj=7.1.0` but not in `proj=7.2.0`

I have version 6.2.1! If version 7.2.0 or above is required, this should be mentioned too.

File: setup.py (hunk 1 above)
Comment (adehecq) [reply setup.py:16]: Also geoutils obviously is a dependency...

**Comment by rhugonnet** [comment]
Oh yes, those tests were still ongoing...
The failure is probably because proj-data is required (not sure the pyproj version changes much). For now, the dependency is only in environment.yml file. Because it is ~500Mo, I wasn't sure it was a good idea to add as a permanent dependency in the setup.py. Maybe we could add a condition to run some of the tests only if proj-data exists?

**Comment by erikmannerfelt** [comment]
> > > I pulled your branch Erik and I also get several errors when running the tests:
> > > ```
> > > * first is that the path to the test data should be set with absolute paths. To check if that works, I suggest installing the package without the -e option and run from outside the package folder. That's when issues start...
> > > 
> > > * in test_vref, I get
> > > ```
> > > 
> > > 
> > > > ```
> > > >       geoid=pyproj.Proj(init="EPSG:4326", geoidgrids='us_nga_egm08_25.tif')
> > > > ```
> > > > 
> > > > 
> > > > tests/test_dem.py:50:
> > > > ../../../miniconda3/envs/work/lib/python3.8/site-packages/pyproj/proj.py:171: in **init**
> > > > super(Proj, self).**init**(cstrencode(projstring.strip()))
> > > > > ???
> > > > > E   pyproj.exceptions.ProjError: Invalid projection b'+proj=longlat +datum=WGS84 +geoidgrids=us_nga_egm08_25.tif +vunits=m +no_defs'.: (Internal Proj Error: proj_create: Error -38: failed to load datum shift file)
> > > > 
> > > > 
> > > > pyproj/_proj.pyx:30: ProjError
> > 
> > 
> > @adehecq what version of proj do you have? I've had the exact same issue using `proj=7.1.0` but not in `proj=7.2.0`
> 
> I have version 6.2.1! If version 7.2.0 or above is required, this should be mentioned too.

@adehecq oops! To be clear, I had the same errors in a different repo, so I only assume that `proj=7.2.0` is the solution.

How does one add a non-python requirement like that? Maybe just a function parsing the result of calling `proj` directly?

**Comment by rhugonnet** [comment]
I was making new commits to account for Amaury's comments on the previous PR before merging. I will move this to another PR!

**Comment by erikmannerfelt** [comment]
> I was making new commits to account for Amaury's comments on the previous PR before merging. I will move this to another PR!

Sorry, @rhugonnet ! Should there be a policy that the author is the only one that should merge, after it has been accepted by someone else?

**Comment by rhugonnet** [comment]
> > I was making new commits to account for Amaury's comments on the previous PR before merging. I will move this to another PR!
> 
> Sorry, @rhugonnet ! Should there be a policy that the author is the only one that should merge, after it has been accepted by someone else

File: setup.py (hunk 1 above)
Comment (erikmannerfelt) [reply setup.py:16]: Added geoutils as a dependency in 6ebdad9

**Comment by rhugonnet** [comment]
Ups wrong button :D! Erik: we're thinking of doing a Zoom with Amaury, it's easier, we'll send the link on Slack!

**Pull request merged**